features/FEATURE_BLE/ble/BLETypes.h

```diff
@@ -324,6 +324,25 @@ struct byte_array_t {
         return _value;
     }
 
+    /**
+     * Returns true if every byte is equal to zero
+     */
+    bool is_all_zeros() {
+        for (size_t i = 0; i < array_size; i++) {
+            if (_value[i] != 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Zero out all bytes
+     */
+    void set_all_zeros() {
```

Comment:
Same comment as above.

Reply:
There already is a external function that does it, it's memset, this is just syntactic sugar and will hopefully get inlined.

Reply:
I'm not concerned about performances; I'm concerned about class responsibilities. 

A shorthand that reset the content of a container to `0` or check if all its bytes are zero is definitely useful. However re-usability with other containers or view to containers is impossible if it's a member function.

I would be happy to write `if (!is_all_zeros(_oob_local_random))` instead of `if (!_oob_local_random.is_all_zeros())` and update `_oob_local_random` type without breaking the code/adding new functions to the new container.